app/src/main/java/com/internship/tvseries/data/repository/auth/AuthRepository.java

```diff
@@ -0,0 +1,7 @@
+package com.internship.tvseries.data.repository.auth;
+
+//Interface for the authentication repository
+public interface AuthRepository {
+    void register(String email, String password);
+    void addAuthenticationSuccessListener(AuthenticationSuccessListener listener);
```

Comment:
You could add the listener as a third parameter of the register method. Please take a look into `Consumer` too.